lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(x)
+    cells[x.to_i-1]
+  end
+
+  def full?
+    cells.all? do |space|
+      space == "X" || space == "O"
+    end
+  end
+
+  def turn_count
+      cells.count{|space| space == "O" || space == "X"}
+  end
+
+  def taken?(location)
+    p = location.to_i-1
+    if cells[p] != " " && cells[p] != ""
+      true
```

Comment:
Avoid returning boolean literals from `if` statements, relying instead on the implicit return value of the boolean expression.
```
def taken?(location)
  p = location.to_i - 1
  cells[o] != " " && cells[p] != ""
end
```